### PR TITLE
Convert cuda/nvshmem api to hip/rocshmem

### PR DESCRIPTION
**Pull request by BKP**

### CODE REVIEW

**Comment by Rombur** [comment]
OK to test

File: src/impl/rocshmemspace/Kokkos_ROCSHMEMSpace.hpp [hunk 1]
```diff
@@ -42,7 +42,7 @@ class ROCSHMEMSpace {
   using execution_space = Kokkos::HIP;
 #else
 #error \
-    "At least the following device execution space must be defined: Kokkos::Cuda."
+    "At least the following device execution space must be defined: Kokkos::Experimental::ROCSHMEMSpace."
```
Comment (dalg24) [inline src/impl/rocshmemspace/Kokkos_ROCSHMEMSpace.hpp:45]:
```suggestion
    "At least the following device execution space must be defined: Kokkos::HIP."
```

File: src/impl/rocshmemspace/Kokkos_ROCSHMEMSpace_AllocationRecord.cpp [hunk 2]
```diff
@@ -176,19 +176,19 @@ SharedAllocationRecord<Kokkos::Experimental::ROCSHMEMSpace, void>::get_record(
 
   // Copy the header from the allocation
   Header head;
-  Header const *const head_cuda =
+  Header const *const head_rocshmem =
```
Comment (dalg24) [inline src/impl/rocshmemspace/Kokkos_ROCSHMEMSpace_AllocationRecord.cpp:179]: Prefer "hip" to 'rochsmem'

Comment (BKP) [reply src/impl/rocshmemspace/Kokkos_ROCSHMEMSpace_AllocationRecord.cpp]: Done.

**Comment by janciesko** [comment]
Please run [apply-clang-format](https://github.com/kokkos/kokkos-remote-spaces/blob/main/scripts/apply-clang-format) to pass CI.

**Comment by BKP** [comment]
> Please run [apply-clang-format](https://github.com/kokkos/kokkos-remote-spaces/blob/main/scripts/apply-clang-format) to pass CI.

Done.

I needed to change the version in the apply-clang-format file from version 8 to version 15 (on Frontier).

There were files in the other directories (shmem, nvshmem) that the apply-clang-format script updated. I only included the files in the rocshmem subdirectory in the pushed changesets.

**Comment by janciesko** [comment]
This still has a small formatting missmatch with our clang-format script.

**Comment by janciesko** [comment]
Let's make this merge contingent on passing unit tests on Crusher

**Comment by BKP** [comment]
Tests pass on both Crusher and Frontier.

**Review by janciesko — APPROVED** [review]

**Pull request merged**